src/main/router.ts

```diff
@@ -140,12 +143,26 @@ export class Router {
         filePath = `${publicPath}/${appName}.html`;
       }
     }
-
   }
 
   protected addRoutes() {
     // Static assets
-    this.router.add({ method: "get", path: "/{path*}" }, Router.handleStaticFile);
+    if (this.dependencies.httpProxy) {
+      this.router.add({ method: "get", path: "/{path*}" }, (apiReq: LensApiRequest) => {
```

Comment:
This is ugly but I didn't want to polish further because it will be refactored anyway in #4859 .